cfgov/regulations3k/scripts/ecfr_importer.py

```diff
@@ -0,0 +1,282 @@
+
+    part = None
+    version = None
+    subparts = {
+        'section_subparts': [],
+        'appendix_subpart': None,
+        'interp_subpart': None,
+    }
+    sections = []
+    appendices = []
+    interpretations = []
+
+
+PAYLOAD = PayLoad()
+LEVEL_STATE = IdLevelState()
+
+
+def parse_part(part_soup, part_number):
+    part, created = Part.objects.get_or_create(
+        part_number=part_number,
+        chapter=CFR_CHAPTER,
+        cfr_title_number=CFR_TITLE)
+    if created:
+        raw_title = part_soup.find('HEAD').text
+        parsed_title = title_pattern.match(raw_title)
+        part.title = parsed_title.group(2).title()
+        part.letter_code = parsed_title.group(3).replace(
+            'REGULATION', '').strip()
+        part.save()
+        logger.info("Created Part {}, {}".format(part_number, part.title))
+    PAYLOAD.part = part
+    return part
+
+
+def parse_version(soup, part):
+    auth = soup.find('AUTH').find('PSPACE').text.strip()
+    source = soup.find('SOURCE').find('PSPACE').text.strip()
+    version, created = EffectiveVersion.objects.get_or_create(
+        acquired=datetime.date.today(),
+        authority=auth,
+        part=part,
+        source=source,
+    )
+    if created:
+        version.draft = True
+        version.save()
+        logger.info("Created new draft version for {}".format(part))
+    PAYLOAD.version = version
+    return version
+
+
+def parse_subparts(part_soup, subpart_list, part):
+    """Createa a mapping of subparts and the sections they contain."""
+    _appendices, cr = Subpart.objects.get_or_create(
+        title="Appendices",
+        label="{}-Appendices".format(part.part_number),
+        version=PAYLOAD.version)
+    PAYLOAD.subparts['appendix_subpart'] = _appendices
+    _interps, cr = Subpart.objects.get_or_create(
+        title="Supplement I to Part {}".format(part.part_number),
+        label="Official Interpretations",
+        version=PAYLOAD.version)
+    PAYLOAD.subparts['interp_subpart'] = _interps
+
+    labeled_subparts = [subp for subp in subpart_list
+                        if subp.find('HEAD').text.strip()]
+    if not labeled_subparts:
+        generic_subpart, cr = Subpart.objects.get_or_create(
+            label=part.part_number,
+            title="General",
+            version=PAYLOAD.version
+        )
+        PAYLOAD.subparts['section_subparts'].append(generic_subpart)
+        parse_sections(part_soup.find_all('DIV8'), part, generic_subpart)
+    else:
+        for element in labeled_subparts:
+            _subpart, cr = Subpart.objects.get_or_create(
+                title=element.find('HEAD').text.strip(),
+                label=part.part_number,
+                version=PAYLOAD.version
+            )
+            PAYLOAD.subparts['section_subparts'].append(_subpart)
+            subpart_sections = element.find_all('DIV8')
+            parse_sections(subpart_sections, part, _subpart)
+
+
+def ital_to_bold(soup):
```

Comment:
This is such a stupid little nitpick that I hesitate to comment, but I think I'd prefer "italic" to "ital".